src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/UpdateContactDetailsContestedMidHandlerTest.java

```diff
@@ -62,7 +68,194 @@ private FinremCallbackRequest buildCallbackRequest() {
             .builder()
             .eventType(EventType.UPDATE_CONTACT_DETAILS)
             .caseDetails(FinremCaseDetails.builder().id(123L).caseType(CONTESTED)
-                .data(FinremCaseData.builder().ccdCaseType(CONTESTED).build()).build())
+                .data(FinremCaseData.builder().ccdCaseType(CONTESTED).contactDetailsWrapper(new ContactDetailsWrapper()).build()).build())
             .build();
     }
-}
\ No newline at end of file
+
+    @Test
+    void givenContestedCase_WhenNotEmptyPostCode_thenHandlerWillShowNoErrorMessage() {
+
+        FinremCallbackRequest finremCallbackRequest = buildCallbackRequest();
+        FinremCaseDetails caseDetails = finremCallbackRequest.getCaseDetails();
+        FinremCaseData data = caseDetails.getData();
+
+        Address address = new Address();
+        address.setPostCode("AB1 1AB");
+        data.getContactDetailsWrapper().setApplicantAddress(address);
+        data.getContactDetailsWrapper().setRespondentAddress(address);
+        data.getContactDetailsWrapper().setApplicantRepresented(YesOrNo.NO);
+        data.getContactDetailsWrapper().setContestedRespondentRepresented(YesOrNo.NO);
+
+        GenericAboutToStartOrSubmitCallbackResponse<FinremCaseData> handle = handler.handle(finremCallbackRequest, AUTH_TOKEN);
+        assertEquals(0, handle.getErrors().size());
+    }
+
+
+    @Test
+    public void givenConsentedCase_WhenEmptyApplicantPostCode_thenHandlerWillShowMessage() {
+
+        FinremCallbackRequest finremCallbackRequest = buildCallbackRequest();
+        FinremCaseDetails caseDetails = finremCallbackRequest.getCaseDetails();
+        FinremCaseData data = caseDetails.getData();
+
+        Address address = new Address();
+        address.setPostCode("");
+        data.getContactDetailsWrapper().setApplicantAddress(address);
+        data.getContactDetailsWrapper().setApplicantRepresented(YesOrNo.NO);
+
+        GenericAboutToStartOrSubmitCallbackResponse<FinremCaseData> handle = handler.handle(finremCallbackRequest, AUTH_TOKEN);
+
+        assertEquals(1, handle.getErrors().size());
+        assertTrue(handle.getErrors().contains("Postcode field is required for applicant address."));
```

Comment:
You should make your unit tests as precise as possible. You can use 
handle.getErrors.get(0) rather than use contains (other instances of this in the other tests)